geos-mesh/src/geos/mesh/doctor/parsing/mesh_stats_parsing.py

```diff
@@ -0,0 +1,202 @@
+from numpy import unique, where
+from geos.mesh.doctor.checks.mesh_stats import Options, Result
+from . import MESH_STATS
+
+__WRITE_STATS = "write_stats"
+__WRITE_STATS_DEFAULT = 0
+
+__OUTPUT = "output"
+
+__DISCONNECTED = "disconnected"
+__DISCONNECTED_DEFAULT = 0
+
+__FIELD_VALUES = "field_values"
+__FIELD_VALUES_DEFAULT = 0
+
+
+def fill_subparser( subparsers ) -> None:
+    p = subparsers.add_parser( MESH_STATS, help=f"Outputs basic properties of a mesh." )
+    p.add_argument( '--' + __WRITE_STATS,
+                    type=int,
+                    required=True,
+                    metavar=[ 0, 1 ],
+                    default=__WRITE_STATS_DEFAULT,
+                    help=( f"\t[int]: The stats of the mesh will be printed in a file" +
+                           " to the folder specified in --output." ) )
+    p.add_argument( '--' + __DISCONNECTED,
+                    type=int,
+                    required=False,
+                    metavar=[ 0, 1 ],
+                    default=__DISCONNECTED_DEFAULT,
+                    help=f"\t[int]: Display all disconnected nodes ids and disconnected cell ids." )
+    p.add_argument( '--' + __FIELD_VALUES,
+                    type=int,
+                    required=False,
+                    metavar=[ 0, 1 ],
+                    default=__FIELD_VALUES_DEFAULT,
+                    help=f"\t[int]: Display all range of field values that seem not realistic." )
+    p.add_argument( '--' + __OUTPUT,
+                    type=str,
+                    required=False,
+                    help=f"[string]: The output folder destination where the stats will be written." )
+
+
+def convert( parsed_options ) -> Options:
+    write_stats = parsed_options[ __WRITE_STATS ]
+    output_folder = parsed_options[ __OUTPUT ]
+    disconnected = parsed_options[ __DISCONNECTED ]
+    field_values = parsed_options[ __FIELD_VALUES ]
+    # input_filepath will be defined in check function before calling __check
+    return Options( write_stats=write_stats,
+                    output_folder=output_folder,
+                    input_filepath="",
+                    disconnected=disconnected,
+                    field_values=field_values )
+
+
+def display_results( options: Options, result: Result ):
+    log_stream = StringIO()
+    stream_handler = logging.StreamHandler( log_stream )
+    stream_handler.setLevel( logging.INFO )
+
+    # Get the root logger and add the StreamHandler to it to possibly output the log to an external file
+    logger = logging.getLogger()
+    logger.addHandler( stream_handler )
+    logger.setLevel( logging.INFO )
+
+    logging.info( f"The mesh has {result.number_cells} cells and {result.number_points} points." )
+    logging.info( f"There are {result.number_cell_types} different types of cells in the mesh:" )
+    for i in range( result.number_cell_types ):
+        logging.info( f"\t{result.cell_types[ i ]}\t\t({result.cell_type_counts[ i ]} cells)" )
```

Comment:
For readability and as you did L80, we could do:
```suggestion
    for cell_type, type_count in zip(result.number_cell_types, result.cell_type_counts ):
        logging.info( f"\t{cell_type}\t\t({type_count} cells)" )
```